libr/io/p/io_zip.c

```diff
@@ -400,6 +404,7 @@ static RIODesc *r_io_zip_open(RIO *io, const char *file, int rw, int mode) {
 							}
 						}
 					}
+					r_list_free(files);
 				}
 			} else {
 				zip_filename = pikaboo + 1;
```

Comment:
just strdup here and remove all the bool logic to know if the ptr is allocated or not